services/inc/service_debug.h

```diff
@@ -16,121 +16,53 @@
-
-/**
- * The debug output function.
- */
-typedef void (*debug_output_fn)(const char *);
+extern LoggerOutputLevel log_compat_level;
+extern debug_output_fn log_compat_callback;
 
-/**
- * Set the debug logger function and optionally the logging level.
- * @param output                The output function. `NULL` to use the existing function.
- * @param debug_output_level    The log level to output.
- */
+// This function sets logging callback and global logging level that are used for binary compatibility
+// with existent applications
 void set_logger_output(debug_output_fn output, LoggerOutputLevel level);
 
-#define LOG_LEVEL_ACTIVE(level)  (log_level_active(level, NULL))
-
-//int log_level_active(LoggerOutputLevel level, void* reserved);
-
-extern void HAL_Delay_Microseconds(uint32_t delay);
-
-// Short Cuts
-#define __LOG_LEVEL_TEST(level) (level >= LOG_LEVEL_AT_COMPILE_TIME && level >= LOG_LEVEL_AT_RUN_TIME)
-
-#if defined(USE_ONLY_PANIC)
-#define LOG(fmt, ...)
-#define DEBUG(fmt, ...)
-#define INFO(fmt, ...)
-#define WARN(fmt, ...)
-#define ERROR(fmt, ...)
-#define PANIC(code,fmt, ...) do {panic_(code, NULL, HAL_Delay_Microseconds);}while(0)
-#define DEBUG_D(fmt, ...)
-#else
-// Macros to use
-#define LOG(fmt, ...)    do { if ( __LOG_LEVEL_TEST(LOG_LEVEL)  )  {log_print_(LOG_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt, ##__VA_ARGS__);}}while(0)
-#define DEBUG(fmt, ...)  do { if ( __LOG_LEVEL_TEST(DEBUG_LEVEL))  {log_print_(DEBUG_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt,##__VA_ARGS__);}}while(0)
-#define INFO(fmt, ...)   do { if ( __LOG_LEVEL_TEST(INFO_LEVEL) )  {log_print_(INFO_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt,##__VA_ARGS__);}}while(0)
-#define WARN(fmt, ...)   do { if ( __LOG_LEVEL_TEST(WARN_LEVEL) )  {log_print_(WARN_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt,##__VA_ARGS__);}}while(0)
-#define ERROR(fmt, ...)  do { if ( __LOG_LEVEL_TEST(ERROR_LEVEL) ) {log_print_(ERROR_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt,##__VA_ARGS__);}}while(0)
-#define PANIC(code,fmt, ...)  do { if ( __LOG_LEVEL_TEST(PANIC_LEVEL) ) {log_print_(PANIC_LEVEL,__LINE__,__PRETTY_FUNCTION__,_FILE_PATH,fmt,##__VA_ARGS__);} panic_(code, NULL, HAL_Delay_Microseconds);}while(0)
-#define DEBUG_D(fmt, ...)  do { if ( __LOG_LEVEL_TEST(DEBUG_LEVEL))  {log_print_direct_(LOG_LEVEL, nullptr, fmt,##__VA_ARGS__);}}while(0)
-#endif
-#define SPARK_ASSERT(predicate) do { if (!(predicate)) PANIC(AssertionFailure,"AssertionFailure "#predicate);} while(0);
+// These functions do nothing. Use alternatives declared in logging.h header instead
+void log_print_(int level, int line, const char *func, const char *file, const char *msg, ...);
+void log_direct_(const char* s);
+void log_print_direct_(int level, void* reserved, const char *msg, ...);
+int log_level_active(LoggerOutputLevel level, void* reserved);
 
 #ifdef __cplusplus
 }
 #endif
 
+#define DEBUG(fmt, ...) LOG(DEBUG, fmt, ##__VA_ARGS__)
```

Comment:
Let's add a comment that these macros are deprecated. This is the compatibility layer.